### ISSUE TITLE
About use_seg

### ISSUE THREAD

**Issue by Davidzhangyuanhan** [facebookresearch/omnivore#16]
I note a _forward_seg_ function in the class _BasicLayer()_ of swin_transformer_3d. 

What does this function work for? 
Should I switch _the forward_ function to  _the forward_seg_ function when I fine-tune for the semantic segmentation task?

Thanks!

**Comment by Davidzhangyuanhan** [comment]
By referring to the https://github.com/SwinTransformer/Swin-Transformer-Semantic-Segmentation, I guess _forward_seg_ works for the semantic segmentation task.

Closed

**Comment by woctezuma** [comment]
This functions is used here:
https://github.com/facebookresearch/omnivore/blob/1e471efe99d6abd04e3855378189b23276c46c0c/models/swin_transformer_3d.py#L558-L565

**Comment by Davidzhangyuanhan** [comment]
thanks! but when the use_seg=True

**Comment by woctezuma** [comment]
We need to wait for an input from the contributors: https://github.com/facebookresearch/omnivore/graphs/contributors

**Comment by Davidzhangyuanhan** [comment]
Thanks.
Referring to https://github.com/SwinTransformer/Swin-Transformer-Semantic-Segmentation/blob/main/mmseg/models/backbones/swin_transformer.py, the code of _forward_seg()_ is the same as the forward function of BasicLayer() in that python file. So I guess that we should set  use_seg=True when I fine-tune for the semantic segmentation task.

**Comment by imisra** [comment]
Hi @Davidzhangyuanhan 

You are correct. `use_seg=True` is set when finetuning on segmentation tasks. We followed the [Swin semantic segmentation repo](https://github.com/SwinTransformer/Swin-Transformer-Semantic-Segmentation.) for implementing this function.

**Comment by Davidzhangyuanhan** [comment]
> 

Thanks for the prompt reply!

But the input tensor size of `forward_seg()` should be (B, H*W, C), which is different from the input tensor size of `forward()` of `BasicLayer()`, i.e., (B, C, **_D_**, H, W). 

Indeed, although the **_D_** = 1 when fine-tuning the segmentation tasks, and we can squeeze that dim followed by the rearranging of the tensor, such implementation is not shown in the code. So I'm a little confused.

**Comment by imisra** [comment]
Yes, that is right. The `use_seg` function was only used in the case of images where D=1 and we sent the re-arranged tensor. The current code is only for classification, and so the squeeze + re-arrange part isn't present.

**Comment by Davidzhangyuanhan** [comment]
Thanks!
